clang/lib/Parse/ParseDeclCXX.cpp

```diff
@@ -4528,6 +4528,54 @@ static bool IsBuiltInOrStandardCXX11Attribute(IdentifierInfo *AttrName,
   }
 }
 
+/// Parse the argument to C++23's [[assume()]] attribute.
+bool Parser::ParseAssumeAttributeArg(ParsedAttributes &Attrs,
```

Comment:
Hopefully someone more familiar with the parsing will come by, my familiarity here is limited.